docs/standards.md

```diff
@@ -0,0 +1,204 @@
+# Brightcove video.js Plugin Standards
+
+When we refer to "standard video.js plugins" we are not referring to any official, codified standard (e.g. ECMAScript or HTML5). Rather, we are referring to the rules used internally at [Brightcove](https://www.brightcove.com) in developing both open-source and proprietary plugins for [video.js](http://videojs.com).
+
+_These rules are by no means required for community plugins._ You can write plugins in whichever way you choose. However, these rules _are recommended_, as we want to foster consistency within the video.js ecosystem.
+
+This document and [the Yeoman generator](https://github.com/videojs/generator-videojs-plugin) it is part of are provided as open-source for the good of the community. If you don't agree with these standards, by all means follow your preferences in your plugin projects!
+
+### Table of Contents
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+
+- [Rules Summary](#rules-summary)
+- [Packaging and Dependencies](#packaging-and-dependencies)
+  - [Structure](#structure)
+- [Automation](#automation)
+  - [npm Core Scripts](#npm-core-scripts)
+- [Coding Style](#coding-style)
+- [Testing](#testing)
+  - [Writing Tests](#writing-tests)
+  - [Testing with Karma](#testing-with-karma)
+  - [Testing in a Browser](#testing-in-a-browser)
+- [Release](#release)
+  - [Versioning](#versioning)
+  - [Publishing](#publishing)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Rules Summary
+
+All standard video.js plugins _must_:
+
+- ...be npm packages.
+- ...have automation available through npm scripts.
+- ...implement the core set of npm scripts.
+- ...be written in ES6 and pass `videojs-standard` linting.
+- ...have tests.
+
+## Packaging and Dependencies
+
+__All standard video.js plugins must be npm packages.__
+
+Plugins should be as self-contained as possible, so the only dependency (`"dependencies"` in `package.json`) by default is video.js, which is shimmed using `browserify-shim`.
+
+Development dependencies (`"devDependencies"` in `package.json`) will include many packages related to developing, building, and testing a standard video.js plugin.
+
+### Structure
```

Comment:
Should we add a column for "generated"? All folders and files that are generators automatically during builds.